sharktank/sharktank/utils/load_llm.py

```diff
@@ -221,23 +221,29 @@ def prefill(self):
             seq_block_ids = [seq_block_ids]
             attention_mask = [attention_mask]
         else:
+            pipeline_to_device_map = (
+                model.config.pipeline_to_device_map
+                if model.config.pipeline_to_device_map
+                else [list(range(shard_count))]
+            )
             token_ids = replicate(
-                token_ids, shard_count, devices=model.config.pipeline_to_device_map[0]
+                token_ids, count=shard_count, devices=pipeline_to_device_map[0]
             )
             _attention_mask, _seq_block_ids = [], []
-            for pipeline in range(model.cache.pipeline_count):
+            for pipeline in range(len(pipeline_to_device_map)):
+                devices = pipeline_to_device_map[pipeline]
```

Comment:
```suggestion
            for pipeline, devices in enumerate(pipeline_to_device_map):
```
This and the next one. Cleans up and already long portion slightly.

Reply:
Done